src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1550,10 +1515,8 @@ private QNode getOrAddNode() {
                         return taskNode;
                     }
                 } else if (headNext instanceof PoolThreadNode || headNext == null) {
-                    if (newNode == null) {
-                        newNode = new PoolThreadNode(Thread.currentThread());
-                    }
                     newNode.setNext(headNext);
```

Comment:
Can this reference become stale and prevent a chain of nodes from getting garbage collected after threads are deallocated?

Reply:
The node might stay around for as long as the thread stays around.  The node *should* be removed from the chain as soon as something is submitted to it, otherwise.  It might be somewhat stale if no tasks are submitted for a while.